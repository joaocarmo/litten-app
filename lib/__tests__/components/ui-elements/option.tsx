import TestRenderer from 'react-test-renderer'
import ThemeProvider from '@components/theme/provider'
import { UIOption } from '@ui-elements'

describe('Snapshot test for the "UISwitch" component', () => {
  const items = [
    {
      key: 'one',
      label: 'One',
      value: 'one',
    },
    {
      key: 'two',
      label: 'Two',
      value: 'two',
    },
  ]
  const selectedValue = 'one'
  it('renders correctly without description', () => {
    const element = TestRenderer.create(
      <ThemeProvider>
        <UIOption
          items={items}
          label="Some text here"
          selectedValue={selectedValue}
        />
      </ThemeProvider>,
    ).toJSON()
    expect(element).toMatchSnapshot()
  })
  it('renders correctly with description', () => {
    const element = TestRenderer.create(
      <ThemeProvider>
        <UIOption
          description="Some more text here"
          items={items}
          label="Some text here"
          selectedValue={selectedValue}
        />
      </ThemeProvider>,
    ).toJSON()
    expect(element).toMatchSnapshot()
  })
})
