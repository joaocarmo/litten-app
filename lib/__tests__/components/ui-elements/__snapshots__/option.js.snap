// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for the "UISwitch" component renders correctly with description 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#c4c4c4",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 12,
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 15.75,
          "fontWeight": "400",
        }
      }
    >
      Some text here
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "200",
          },
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      One
    </Text>
  </View>,
  <Text
    style={
      Array [
        Array [
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "200",
          },
          Object {
            "marginTop": 8,
          },
        ],
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Some more text here
  </Text>,
  <RNCAndroidDialogPicker
    items={
      Array [
        Object {
          "label": "One",
          "value": "one",
        },
        Object {
          "label": "Two",
          "value": "two",
        },
      ]
    }
    mode="dialog"
    onSelect={[Function]}
    selected={0}
    style={
      Array [
        Object {
          "height": 50,
        },
        Object {
          "height": 2000,
          "position": "absolute",
          "top": 0,
          "width": 2000,
        },
      ]
    }
  />,
]
`;

exports[`Snapshot test for the "UISwitch" component renders correctly without description 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#c4c4c4",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 12,
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 15.75,
          "fontWeight": "400",
        }
      }
    >
      Some text here
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "200",
          },
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      One
    </Text>
  </View>,
  <RNCAndroidDialogPicker
    items={
      Array [
        Object {
          "label": "One",
          "value": "one",
        },
        Object {
          "label": "Two",
          "value": "two",
        },
      ]
    }
    mode="dialog"
    onSelect={[Function]}
    selected={0}
    style={
      Array [
        Object {
          "height": 50,
        },
        Object {
          "height": 2000,
          "position": "absolute",
          "top": 0,
          "width": 2000,
        },
      ]
    }
  />,
]
`;
